src/explore-education-statistics-common/src/modules/table-tool/utils/__tests__/mapTableToJson.test.ts

```diff
@@ -515,134 +548,1845 @@ describe('mapTableToJson', () => {
     ]);
   });
 
-  test('correctly adds groups across three levels of column headers with same labels', () => {
-    const result = mapTableToJson({
-      tableHeadersConfig: testTableWithThreeLevelsOfColHeadersWithMultipleGroupsWithSameLabelsConfig,
-      subjectMeta:
-        testTableWithThreeLevelsOfColHeadersWithMultipleGroupsWithSameLabels.subjectMeta,
-      results:
-        testTableWithThreeLevelsOfColHeadersWithMultipleGroupsWithSameLabels.results,
-    }).tableJson;
+  describe('Handles multiple filter groups with the same labels', () => {
```

Comment:
Probably worth keeping these in now that we have them, but I'd suggest we try and add some extra cases to `createExpandedColumnHeaders` and `createExpandedRowHeaders` test suites to cover these cases as well.

It's also much easier to setup test data for these suites, so it would be good if we could test some more complex variations as well (if possible).

Reply:
I've added extra test cases for these to `createExpandedColumnHeaders` and `createExpandedRowHeaders`. I actually find it a bit easier testing the tables in `mapTableToJson` because it's simple to try out the test data in the actual app and see what the tables look like, which is harder to do with just the row/col headers. We could definitely do with a better way to generate test data though as it's a faff either way.

I'm struggling to come up with other valid more complex iterations to test, though I'm sure there must be some! If you have any specific ones you think should be tested let me know and I'll add them.

Reply:
There's probably lots to explore around variations of `returns correct headers with deeply nested rows and multiple identical headers` from `createExpandedColumnHeaders`.

At they very least I'd add this test case to `createExpandedRowHeaders` as well.

We could also write up some random-ish permutations of this test case (for both column and row headers) and label them like `returns correct headers for complex case X` (where X is just the test case number e.g. 1, 2, 3, etc).

It'd be nice if there was a way to test permutation in a more complete way. We could probably  write a utility function to generate complete sets of every header tree permutation (up to a limit), then we'd just need to run them against some snapshots.

Reply:
I've added tests for 'multiple filter groups with the same labels' for `createExpandedColumnHeaders` and `createExpandedRowHeaders`  now (totally thought I had already, but it was ones for the rows of all merged cells that I added).

I'm not sure there's a lot of value to manually adding more permutations to the tests, there are already a lot of cases covered between the three test files, especially around duplicate groups and merged cells which have been the problem areas.